streams/src/test/java/org/apache/kafka/streams/integration/IQv2StoreIntegrationTest.java

```diff
@@ -1071,16 +1568,51 @@ public <V> void shouldHandleKeyQuery(
 
         assertThrows(IllegalArgumentException.class, queryResult::getFailureReason);
         assertThrows(
-                IllegalArgumentException.class,
-                queryResult::getFailureMessage
+            IllegalArgumentException.class,
+            queryResult::getFailureMessage
         );
 
         final V result1 = queryResult.getResult();
         final Integer integer = valueExtactor.apply(result1);
         assertThat(integer, is(expectedValue));
         assertThat(queryResult.getExecutionInfo(), is(empty()));
         assertThat(queryResult.getPosition(), is(POSITION_0));
-    }   
+    }
+
+    public <V> void shouldHandleKeyPAPIQuery(
```

Comment:
Maybe I missed it, but how is this method different to `shouldHandleKeyDSLQuery` ? -- Do we really need to versions of exiting `shouldHandleKeyQuery` ?